imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js

```diff
@@ -114,6 +133,74 @@ Template.variantForm.helpers({
         });
       };
     };
+  },
+  isProviderEnabled: function () {
+    const shopId = Reaction.getShopId();
+
+    const provider = Packages.findOne({
+      "shopId": shopId,
+      "registry.provides": "taxCodes",
+      "$where": function () {
+        const providerName = this.name.split("-")[1];
+        return this.settings[providerName].enabled;
+      }
+    });
+
+    if (provider) {
+      return true;
+    }
+  },
+  listTaxCodes: function () {
+    const instance = Template.instance();
+    const shopId = Reaction.getShopId();
+
+    const provider = Packages.findOne({
+      "shopId": shopId,
+      "registry.provides": "taxCodes",
+      "$where": function () {
+        const providerName = this.name.split("-")[1];
+        return this.settings[providerName].enabled;
+      }
+    });
+
+    if (provider) {
+      if (Meteor.subscribe("TaxCodes").ready() && TaxCodes.find({}).count() === 0) {
+        Meteor.call(provider.settings.taxCodes.getTaxCodeMethod, (error, result) => {
+          result.forEach(function (code) {
```

Comment:
@kieha Here this function does not check that result is an Array nor does it check for an error before trying to iterate over it.